qemu/tests/uefi_check_debugcon.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import env_process
+from virttest import error_context
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the "-debugcon" parameter under the UEFI environment:
+    1) Boot up a guest.
+       If params["ovmf_log"] is not None,
+       append debugcon parameter to qemu command lines.
+    2) Remove the existing isa-log device.
+    3) Destroy the guest.
+    4) Start the trace command on host.
+    5) Re-create the guest and verify it is alive.
+    6) Destroy the guest.
+    7) Check pio_read counts and pio_write counts.
+    7.1) If disable debugcon:
+            pio_read_counts > 0
+            pio_write_counts = 0
+    7.2) If enable debugcon:
+            pio_read_counts > 0
+            pio_write_counts > 0
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def check_trace_process():
+        """
+        check whether trace process is existing
+        """
+        if process.system(
+                params["grep_trace_cmd"], ignore_status=True, shell=True):
+            return False
+        else:
+            return True
+
+    def remove_isa_debugcon(vm):
+        """
+        remove the existing isa-log device
+        """
+        for device in vm.devices:
+            if device.type == "isa-log":
+                vm.devices.remove(device)
+                env.register_vm(vm.name, vm)
```

Comment:
Hi, no need to register VM in loop, this line should be in out-side of for loop

Reply:
updated, thanks.